chain/client/src/stateless_validation/chunk_validator/mod.rs

```diff
@@ -258,10 +255,18 @@ impl Client {
             "process_chunk_state_witness",
         );
 
+        // Chunk producers should not receive state witness from themselves.
+        log_assert!(
+            signer.is_some(),
+            "Received a chunk state witness but this is not a validator node. Witness={:?}",
+            witness
+        );
+        let signer = signer.unwrap();
+
```

Comment:
not your change but this is iffy
1) The comment doesn't match the code. 
2) Can an attacker send state witness to an RPC node and crash it? 

Best to handle it outside of this PR though. 
cc @pugachAG

Reply:
> Can an attacker send state witness to an RPC node and crash it?

I don't this so, this code should only be executed after the witness was validated as part of partial witness actor, so this is safe